components/core/src/clp/ffi/ir_stream/IrErrorCode.cpp

```diff
@@ -0,0 +1,25 @@
+#include "IrErrorCode.hpp"
+
+#include <string>
+
+using IrErrorCategory = clp::error_handling::ErrorCategory<clp::ffi::ir_stream::IrErrorCodeEnum>;
+using clp::ffi::ir_stream::IrErrorCodeEnum;
+
+template <>
+auto IrErrorCategory::name() const noexcept -> char const* {
+    return "clp::ffi::ir_stream::IrErrorCode";
+}
+
+template <>
+auto IrErrorCategory::message(IrErrorCodeEnum error_enum) const -> std::string {
+    switch (error_enum) {
+        case IrErrorCodeEnum::DecodingMethodFailure:
+            return "Decoding method failure.";
+        case IrErrorCodeEnum::EndOfStream:
+            return "End-of-stream has been reached.";
+        case IrErrorCodeEnum::IncompleteStream:
+            return "Incomplete IR stream.";
```

Comment:
Lets try to be a bit more descriptive for these messages, since right now they just repeat the enum name for the most part. Ideally, we want the user to have some idea on where to look to understand the source of the error.

For example, for `EndOfStream` maybe:
"The end-of-stream encoding tag has been consumed."

I'm not sure that is very good, but I think if we discuss for a bit we can come up with some better stuff.